### PR TITLE
Graviton monitoring differences

### PR DESCRIPTION
**Pull request by RamaMalladiAWS**
This change request highlights some differences in utilities and tools available for monitoring on AWS Graviton processors compared to x86 processor architectures.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Graviton_monitoring_differences.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+|*turbostat* support	|Yes	|No	|
+|*i7z* Works	|Yes	|No	|
+|Performance monitoring tools	|_[VTune Profiler](https://www.intel.com/content/www/us/en/developer/tools/oneapi/vtune-profiler.html) and [PCM](https://github.com/opcm/pcm), [Linux perf](https://www.brendangregg.com/perf.html)_	|_[Linux perf](https://www.brendangregg.com/perf.html), [Arm Forge](https://developer.arm.com/Tools%20and%20Software/Arm%20Forge)_	|
+
+More details on how to use Linux perf utility on AWS Graviton processors is available [here](https://github.com/aws/aws-graviton-getting-started/blob/main/optimizing.md#profiling-the-code).
+
+**Note:** On AWS Graviton 2/3 processors, Turbo isn’t supported. So, utilities such as ‘turbostat’ aren’t supported/ relevant for Arm architecture (and not on AWS Graviton processor either). Also, tools such as *[i7z](https://code.google.com/archive/p/i7z/)* for discovering CPU frequency, turbo, sockets and other information are only supported on Intel architecture/ processors.
+
+## Sample output
+#### Using lscpu utility
```
Comment (snadampal) [inline Graviton_monitoring_differences.md:17]: I feel we are missing the flow... not sure why is lscpu mentioned here. we can say something like, lscpu can be used to get architecture details, cache levels and sizes, and HW capabilities via flags. However, it doesn't provide the cpu clock for arm/aarch64 systems for which we can use "dmidecode". then talk about "dmidecode'': how to run and what to look for etc...

File: Graviton_monitoring_differences.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+|Socket Designation| CPU00|
+|Type| Central Processor|
+|Family| ARMv8|
+|Manufacturer| AWS|
+|...|...|
+|Max Speed| 2600 MHz|
+|Current Speed| 2600 MHz|
+|...|...|
+
+#### Collect CPU counter statistics for the specified command
```
Comment (snadampal) [inline Graviton_monitoring_differences.md:42]: again, introduce perf and talk about the commands.

File: Graviton_monitoring_differences.md (hunk 1 above)
Comment (RamaMalladiAWS) [reply Graviton_monitoring_differences.md:17]: We can explicitly mention that in words or just put it in a table. So, I put the list of tools in a table to summarize quickly. The idea is that CPU frequency can be listed by utilities such as lscpu for x86 architectures but not for Graviton. So, for that we use dmidecode.

File: Graviton_monitoring_differences.md (hunk 2 above)
Comment (RamaMalladiAWS) [reply Graviton_monitoring_differences.md:42]: I think perf is introduced in the perfrunbook page and also on Brendan Gregg perf page. So, I kept it brief here.

Comment (AGSaidi) [reply Graviton_monitoring_differences.md:42]: still, we don't have things like intel's PCM and we need to say you can use perf and it has these optios

Comment (AGSaidi) [reply Graviton_monitoring_differences.md:42]: same for intel memory latency checker, lmbench or other

Comment (AGSaidi) [reply Graviton_monitoring_differences.md:42]: please talk about how to use this for someone coming from other tools and what perf can measure. e.g. on metal you can get interconnect counters too.

**Comment by AGSaidi** [comment]
please add hwloc and lstopo

**Comment by RamaMalladiAWS** [comment]
> please add hwloc and lstopo

I have added these to the doc.

File: Graviton_monitoring_differences.md (hunk 2 above)
Comment (RamaMalladiAWS) [reply Graviton_monitoring_differences.md:42]: I prefect not to add lmbench and Intel memory latency chencker as the list is a bit unending and we can't cover all the tools on such a page. By definition, the Intel MLC is an Intel tool, so I don't see a point in adding it here. I tried to build lmbench but with some dependency failures and I feel it isn't worth it.

I have added details on how to use Linux perf and a note about PCM.

Comment (RamaMalladiAWS) [reply Graviton_monitoring_differences.md:42]: With some fixes/ updates to the package, I could lmbench to build and run successfully. So, there isn't anything unique about it to compare a run on x86 vs Graviton and to document it here.

**Comment by RamaMalladiAWS** [comment]
I have updated the page with the details suggested/ feedback provided.

File: Monitoring_Tools_on_Graviton.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
+# Monitoring tools for AWS Graviton
+Listed below are some monitoring and profiling tools supported on AWS Gravtion. Also listed are some differences when compared to the tools available on x86 processor architectures.
+
+Some of the most commonly used tools such as _top, htop, iostat, lstopo, hwloc, dmidecode, lmbench, Linux perf_ are supported on AWS Graviton processors. There are some tools such as Intel MLC, Intel VTune Profiler, PCM, _turbostat_ are only supported on Intel processors.
```
Comment (AGSaidi) [inline Monitoring_Tools_on_Graviton.md:4]: x86? turbostat works on amd

File: Monitoring_Tools_on_Graviton.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+|*hwloc* support	|Yes	|Yes	|
+|*lstopo* support	|Yes	|Yes	|
+|*i7z* Works	|Yes	|No	|
+|*lmbench*	|Yes	|Yes	|
+|Intel *MLC*  |Yes    |No     |
+|Performance monitoring tools	|_[VTune Profiler](https://www.intel.com/content/www/us/en/developer/tools/oneapi/vtune-profiler.html) and [PCM](https://github.com/opcm/pcm), [Linux perf](https://www.brendangregg.com/perf.html)_	|_[Linux perf](https://www.brendangregg.com/perf.html), [Arm Forge](https://developer.arm.com/Tools%20and%20Software/Arm%20Forge)_	|
+
+Utilities such as *lmbench* are available [here](http://lmbench.sourceforge.net/) and can be built for AWS Graviton processors to obtain latency and bandwidth stats.
+
+**Note:** On AWS Graviton 2/3 processors, Turbo isn’t supported. So, utilities such as ‘turbostat’ aren’t supported/ relevant for Arm architecture (and not on AWS Graviton processor either). Also, tools such as *[i7z](https://code.google.com/archive/p/i7z/)* for discovering CPU frequency, turbo, sockets and other information are only supported on Intel architecture/ processors. Intel *MLC* is a memory latency checker utility that is only supported on Intel processors.
```
Comment (AGSaidi) [inline Monitoring_Tools_on_Graviton.md:124]: explictly call out the arm linux kernel community has decided not to put cpu frequency in /proc/cpuinfo which can be read by tools such as lscpu or directly?

File: Monitoring_Tools_on_Graviton.md (hunk 3 above)
Comment (RamaMalladiAWS) [reply Monitoring_Tools_on_Graviton.md:4]: Agree... will change

File: Monitoring_Tools_on_Graviton.md (hunk 4 above)
Comment (RamaMalladiAWS) [reply Monitoring_Tools_on_Graviton.md:124]: Ok, wasn't sure to put that in... I will add.

**Review by AGSaidi — APPROVED** [review]

**Pull request merged**